extension/ext-plugin/src/test/plugin.test.ts

```diff
@@ -0,0 +1,159 @@
+      // LaTex single line comment
+      assert.deepEqual(plugin.removeCommentChar('% howdoi query', commentChar), 'howdoi query')
+      // W/o whitespace
+      assert.deepEqual(plugin.removeCommentChar('%howdoi query', commentChar), 'howdoi query')
+    })
+    it('Removal of comment char: ;', function () {
+      const commentChar = { frontComment: ';', endComment: '' }
+      // clojure single line comment
+      assert.deepEqual(plugin.removeCommentChar('; howdoi query', commentChar), 'howdoi query')
+      // W/o whitespace
+      assert.deepEqual(plugin.removeCommentChar(';howdoi query', commentChar), 'howdoi query')
+    })
+  })
+
+  describe('Removal of howdoi prefix test -> #removeHowdoiPrefix', function () {
+    it('Normal Query', function () {
+      // Normal Query
+      assert.equal(plugin.removeHowdoiPrefix('howdoi query'), 'query')
+    })
+    it('Query with whitespace', function () {
+      // Query with whitespace
+      assert.equal(plugin.removeHowdoiPrefix(' howdoi query '), 'query')
+    })
+    it('Query without howdoi prefix', function () {
+      // Query without howdoi prefix
+      assert.equal(plugin.removeHowdoiPrefix('query'), 'query')
+    })
+    it('Query without howdoi prefix and whitespace', function () {
+      // Query without howdoi prefix and whitespace
+      assert.equal(plugin.removeHowdoiPrefix(' query '), 'query')
+    })
+  })
+
+  describe('Add comment character to a string -> #addComment', function () {
+    it('Add comment with front/back chars to string', function () {
+      // Add comment: '/*' '*/'
+      const commentChar1 = { frontComment: '/*', endComment: '*/' }
+      assert.equal(plugin.addComment('howdoi query', commentChar1), '/* howdoi query */')
+      // Add comment with '<!--' '-->' 
+      const commentChar2 = { frontComment: '<!--', endComment: '-->' }
+      assert.equal(plugin.addComment('howdoi query', commentChar2), '<!-- howdoi query -->')
+    })
+    it('Add comment with front char to string', function () {
+      // Add comment: #
+      const commentChar1 = { frontComment: '#', endComment: '' }
+      assert.equal(plugin.addComment('howdoi query', commentChar1), '# howdoi query')
+      // Add comment: ;
+      const commentChar2 = { frontComment: ';', endComment: '' }
+      assert.equal(plugin.addComment('howdoi query', commentChar2), '; howdoi query')
+      // Add comment: %
+      const commentChar3 = { frontComment: '%', endComment: '' }
+      // LaTex single line comment
+      assert.equal(plugin.addComment('howdoi query', commentChar3), '% howdoi query')
+      // Add coment: -- 
+      const commentChar4 = { frontComment: '--', endComment: '' }
+      assert.equal(plugin.addComment('howdoi query', commentChar4), '-- howdoi query')
+      // Add comment: //
+      const commentChar5 = { frontComment: '//', endComment: '' }
+      assert.equal(plugin.addComment('howdoi query', commentChar5), '// howdoi query')
+    })
+  })
```

Comment:
Good use of tests here. As a potential future improvement, you could perhaps generate the test string from the `commentChar` object, e.g.

`{ frontComment: '//', endComment: '' }` -> `'// howdoi query'`

And then have a loop over the `commentChar` objects.